oblib/parser.py

```diff
@@ -279,22 +387,42 @@ def from_XML_string(self, xml_string, entrypoint_name=None):
             if axis is not None:
                 for a in axis:
                     kwargs[a] = axis[a]
-            dm_ctx = data_model.Context(**kwargs)
-            contexts[context.attrib["id"]] = dm_ctx
+
+            if instant is None  and duration is None:
+                validation_errors.append(
+                    ValidationError("Context is missing both a duration and instant tag"))
+            if entity is None:
+                validation_errors.append(ValidationError("Context is missing an entity tag"))
+
+            try:
+                dm_ctx = data_model.Context(**kwargs)
+                contexts[context.attrib["id"]] = dm_ctx
+            except Exception as e:
+                validation_errors.append(ValidationError(str(e)))
 
         # Read all elements that are not a context or a unit:
         for child in root:
             if child.tag != _xn("link:schemaRef") and child.tag != _xn("unit") and child.tag != _xn("context"):
                 kwargs = {}
                 if "contextRef" in child.attrib:
-                    kwargs["context"] = contexts[child.attrib["contextRef"]]
-
-                tag = child.tag
-                tag = tag.replace("{http://xbrl.us/Solar/v1.2/2018-03-31/solar}", "solar:")
-                tag = tag.replace("{http://fasb.org/us-gaap/2017-01-31}", "us-gaap:")
-                tag = tag.replace("{http://xbrl.sec.gov/dei/2014-01-31}", "dei:")
+                    if child.attrib["contextRef"] in contexts:
+                        kwargs["context"] = contexts[child.attrib["contextRef"]]
+                        tag = child.tag
+                        tag = tag.replace("{http://xbrl.us/Solar/v1.2/2018-03-31/solar}", "solar:")
+                        tag = tag.replace("{http://fasb.org/us-gaap/2017-01-31}", "us-gaap:")
+                        tag = tag.replace("{http://xbrl.sec.gov/dei/2014-01-31}", "dei:")
+                        try:
+                            entrypoint.set(tag, child.text, **kwargs)
+                        except Exception as e:
+                            validation_errors.append(ValidationError(str(e)))
+                    else:
+                        validation_errors.append("referenced context is missing")
+                else:
+                    validation_errors.append("Element is missing a context")
 
-                entrypoint.set(tag, child.text, **kwargs)
+        # Raise the errors if necessary
+        if len(validation_errors.get_errors()) > 0:
```

Comment:
`if validation_errors.get_errors():` is enough, no need for `len`

Reply:
Will fix